### DISCUSSION TITLE
Error while creating executable file from a python folder(made from auto_py_to_exe)

### DISCUSSION THREAD

**Discussion by Rishavshahi1702** [pyinstaller/pyinstaller#7642] [Help]
Unable to compile the code in an executable because of a unhandled exception.I am not sure what exactly is happening behind the curtains.Any help is highly appriciated.

PS C:\Users\A5642\Desktop\New folder (2)\VY_final_db_TP_copy_VY> .\code_v5.exe
Traceback (most recent call last):
  File "code_v5.py", line 89, in <module>
    event, values = window_f2.read()
  File "PySimpleGUI\PySimpleGUI.py", line 10065, in read
  File "PySimpleGUI\PySimpleGUI.py", line 22969, in _refresh_debugger
  File "inspect.py", line 1554, in stack
  File "inspect.py", line 1531, in getouterframes
  File "inspect.py", line 1501, in getframeinfo
  File "PyInstaller\hooks\rthooks\pyi_rth_inspect.py", line 22, in _pyi_getsourcefile
**AttributeError: 'function' object has no attribute 'getfile'**
[11512] Failed to execute script 'code_v5' due to unhandled exception!

**Comment by rokm** [comment]
Do you assign a variable called `inspect` somewhere in `code_v5.py`?

**Reply by rokm** [reply]
Or maybe have a function named `inspect` in global namespace?

**Reply by Rishavshahi1702** [reply]
Hi..No we haven't used inspect(as inspect or inspect)

**Reply by rokm** [reply]
Hmm... this still looks like something or someone overrode the `inspect` in the global namespace with a function...

Can you try opening `PyInstaller/hooks/rthooks/pyi_rth_inspect.py` in your local PyInstaller installation, and add `import inspect` at the top of the `def _pyi_getsourcefile(object):`, i.e., change

https://github.com/pyinstaller/pyinstaller/blob/4b22d09c886542f3beea3d23d6a3816af38986de/PyInstaller/hooks/rthooks/pyi_rth_inspect.py#L21-L23

into 

```python
def _pyi_getsourcefile(object):
    import inspect
    filename = inspect.getfile(object)
    if not os.path.isabs(filename):
```

Does that fix the problem?
